### ISSUE TITLE
Tag v0.7

### ISSUE THREAD

**Issue by dlfivefifty** [JuliaApproximation/FastTransforms.jl#82]
ApproxFun/SingularIntegralEquations.jl tests pass! 

@MikaelSlevinsky Happy for me to tag?

**Comment by dlfivefifty** [comment]
Can we wrap this up so it can be tagged?

**Comment by dlfivefifty** [comment]
Would it be easier to drop Appveyor and use windows on Travis?

**Comment by MikaelSlevinsky** [comment]
Sorry, at one point this was my primary focus. Since then it has drifted away. Here are the issues:

- Travis Windows support does not contain Vcpkg. Instead it has Chocolatey which does not have the requisite dependencies openBLAS or FFTW.
- Vcpkg on Appveyor works wonderfully when using a cache. But without a cache, it does not install binaries like homebrew or apt-get. Therefore, openBLAS takes about 20 minutes to build. One impractical fix would require all downstream packages to store a cache. A more practical fix would use the BinaryBuilder infrastructure.
- But BinaryBuilder is support is held up by the following issue https://github.com/JuliaPackaging/BinaryBuilder.jl/issues/442, which I suspect won't be addressed until a major refactoring of the project and build workflow https://github.com/JuliaPackaging/BinaryBuilder.jl/pull/441, which from the referred blog post seems to be a substantial improvement for JLL packages with dependencies!
- A patch to "move forward" would be to drop windows testing and try to add it again on a subsequent tag. With this option, we may find out exactly how many Windows users there are.

Basically, I'm confident that with a `dlopen`able dll file in the deps folder, the Windows tests would pass. This is because the C tests pass in Windows and the Julia tests are quite basic.

**Comment by dlfivefifty** [comment]
You may be surprised how many extant windows users are out there... but I guess they can use old versions of ApproxFun until this is fixed.

Though how does SpecialFunctions.jl do this?

**Comment by MikaelSlevinsky** [comment]
SpecialFunctions uses BinaryBuilder for the tar balls of openspecfun stored in a release of Yggdrasil (https://github.com/JuliaMath/SpecialFunctions.jl/blob/master/deps/build.jl). FastTransforms has an outstanding issue with BinaryBuilder (see above).

**Comment by MikaelSlevinsky** [comment]
Hmm, it seems as though the refactor of BinaryBuilder is already in progress, with the new organization https://github.com/JuliaBinaryWrappers

**Comment by giordano** [comment]
I don't think the next version of BinaryBuilder is going to be officially released anytime soon, but you can already get tarballs built with it, or use the PR yourself.  The best way is to open an issue or a pull request in https://github.com/JuliaPackaging/Yggdrasil

**Comment by dlfivefifty** [comment]
Is it ready to be tagged now?

**Comment by MikaelSlevinsky** [comment]
This is as ready as it gets. The lines that must be added to downstream CI scripts are relatively small. The first time building in Appveyor is slow because `vcpkg` builds dependencies (and OpenBLAS takes a while). But if one adds the `vcpkg` directory to the cache, then subsequent commit tests are faster than Linux builds.

Of course I would love to get BinaryBuilder working for this repository, but I think that's easier said than done. (See https://github.com/JuliaPackaging/BinaryBuilder.jl/issues/442).

**Comment by giordano** [comment]
> Of course I would love to get BinaryBuilder working for this repository

Do you have a builder for this?

**Comment by dlfivefifty** [comment]
Done!

**Comment by MikaelSlevinsky** [comment]
@giordano, all the issues that I've encountered are mentioned in the link above. (The script in the issue could be updated to the C lib's release v0.2.9, but that doesn't change the issues.)

Mainly, Julia's gcc doesn't ship with compiler support libraries. Julia v1.3's artifacts present an elegant solution; however, when I tried `using CompilerSupportLibraries_jll`, it appears broken.

**Comment by giordano** [comment]
> when I tried using CompilerSupportLibraries_jll, it appears broken.

```julia
julia> using CompilerSupportLibraries_jll
[ Info: Precompiling CompilerSupportLibraries_jll [e66e0078-7015-5450-92f7-15fbd957f2ae]

julia>
```

**Comment by MikaelSlevinsky** [comment]
What am I doing wrong?
```julia
Richards-MBP-2:~ Mikael$ julia-1.3
               _
   _       _ _(_)_     |  Documentation: https://docs.julialang.org
  (_)     | (_) (_)    |
   _ _   _| |_  __ _   |  Type "?" for help, "]?" for Pkg help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 1.3.0-rc4.1 (2019-10-15)
 _/ |\__'_|_|_|\__'_|  |  Official https://julialang.org/ release
|__/                   |

julia> versioninfo()
Julia Version 1.3.0-rc4.1
Commit 8c4656b97a (2019-10-15 14:08 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin18.6.0)
  CPU: Intel(R) Core(TM) i7-4980HQ CPU @ 2.80GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.1 (ORCJIT, haswell)

(v1.3) pkg> status
    Status `~/.julia/environments/v1.3/Project.toml`
  [12aac903] BinaryBuilder v0.1.4 #master (https://github.com/JuliaPackaging/BinaryBuilder.jl.git)
  [e66e0078] CompilerSupportLibraries_jll v0.2.0+0 #master (https://github.com/JuliaBinaryWrappers/CompilerSupportLibraries_jll.jl.git)

julia> using CompilerSupportLibraries_jll
ERROR: InitError: could not load library "/Users/Mikael/.julia/artifacts/0f0c9ee55a39f0c6ccc50e07baf798d64e0209cb/lib/libgfortran.5.dylib"
dlopen(/Users/Mikael/.julia/artifacts/0f0c9ee55a39f0c6ccc50e07baf798d64e0209cb/lib/libgfortran.5.dylib, 1): Library not loaded: /workspace/destdir/x86_64-apple-darwin14/lib/libquadmath.0.dylib
  Referenced from: /Users/Mikael/.julia/artifacts/0f0c9ee55a39f0c6ccc50e07baf798d64e0209cb/lib/libgfortran.5.dylib
  Reason: image not found
Stacktrace:
 [1] #dlopen#3(::Bool, ::typeof(Libdl.dlopen), ::String, ::UInt32) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.3/Libdl/src/Libdl.jl:109
 [2] dlopen at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.3/Libdl/src/Libdl.jl:109 [inlined] (repeats 2 times)
 [3] __init__() at /Users/Mikael/.julia/packages/CompilerSupportLibraries_jll/psVxF/src/wrappers/x86_64-apple-darwin14-libgfortran5.jl:91
 [4] _include_from_serialized(::String, ::Array{Any,1}) at ./loading.jl:692
 [5] _require_search_from_serialized(::Base.PkgId, ::String) at ./loading.jl:776
 [6] _require(::Base.PkgId) at ./loading.jl:1001
 [7] require(::Base.PkgId) at ./loading.jl:922
 [8] require(::Module, ::Symbol) at ./loading.jl:917
during initialization of module CompilerSupportLibraries_jll

julia> 
```

**Comment by giordano** [comment]
Not sure, but I'm using GNU/Linux, not macOS.  If you submit the builder to [Yggdrasil](https://github.com/JuliaPackaging/Yggdrasil/) you don't need to build it yourself (even though would be good to resolve this issue anyway)

**Comment by giordano** [comment]
@MikaelSlevinsky I think I miss why you want to use `CompilerSupportLibraries_jll` outside of the BinaryBuilder environment

**Comment by MikaelSlevinsky** [comment]
I was under the impression that an end user would still require the dynamic libraries that this one depends on. With the artifact system, this would mean that a FastTransforms_jll would depend on MPFR_jll, OpenBLAS_jll, CompilerSupportLibraries_jll (for libgomp), and FFTW_jll, even after compilation of the FastTransforms library itself. Please let me know if this is not the case.

Cheers,

Mikael

On Nov 9, 2019, at 7:59 AM, Mosè Giordano <notifications@github.com> wrote:

﻿

@MikaelSlevinsky<https://github.com/MikaelSlevinsky> I think I miss why you want to use CompilerSupportLibraries_jll outside of the BinaryBuilder environment

**Comment by giordano** [comment]
Julia already comes with MPFR and OpenBLAS, so there is no need to install them on the user machine.  My understanding of `CompilerSupportLibraries_jll` is that it provides in the BinaryBuilder environment only system libraries that are supposed to already be on the user's system.  Again, there should be no need to install it.

FFTW is different, as it doesn't come with Julia and it's not supposed to be already available.

I encourage once more to push your builder to Yggdrasil, we now have a CI system that automatically builds the tarballs, see for example [this build](https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=48), for https://github.com/JuliaPackaging/Yggdrasil/pull/164

**Comment by MikaelSlevinsky** [comment]
> My understanding of CompilerSupportLibraries_jll is that it provides in the BinaryBuilder environment only system libraries that are supposed to already be on the user's system.

I'm not sure that's true. Consider locating `libgomp.1.dylib` by installing a vanilla Julia-1.3 (release candidate 4.1).

`Libgomp` is not shipped with Julia-1.3:
```
sec-12-0380:lib Mikael$ cd /Applications/Julia-1.3.app/Contents/Resources/julia/lib/julia/
sec-12-0380:julia Mikael$ ls
libLLVM-6.0.dylib			libmbedtls.2.16.0.dylib
libLLVM.dylib				libmbedtls.dylib
libamd.2.4.6.dylib			libmbedx509.0.dylib
libamd.2.dylib				libmbedx509.2.16.0.dylib
libamd.dylib				libmbedx509.dylib
libcamd.2.4.6.dylib			libmpfr.6.dylib
libcamd.2.dylib				libmpfr.dylib
libcamd.dylib				libopenblas64_.0.3.5.dylib
libccalltest.dylib			libopenblas64_.dylib
libccolamd.2.9.6.dylib			libopenlibm.2.5.dylib
libccolamd.2.dylib			libopenlibm.2.dylib
libccolamd.dylib			libopenlibm.dylib
libcholmod.3.0.13.dylib			libpcre2-8.0.dylib
libcholmod.3.dylib			libpcre2-8.dylib
libcholmod.dylib			libquadmath.0.dylib
libcolamd.2.9.6.dylib			libspqr.2.0.9.dylib
libcolamd.2.dylib			libspqr.2.dylib
libcolamd.dylib				libspqr.dylib
libcurl.4.dylib				libssh2.1.0.1.dylib
libcurl.dylib				libssh2.1.dylib
libdSFMT.dylib				libssh2.dylib
libgcc_s.1.dylib			libsuitesparse_wrapper.dylib
libgfortran.5.dylib			libsuitesparseconfig.5.4.0.dylib
libgit2.0.28.2.dylib			libsuitesparseconfig.5.dylib
libgit2.28.dylib			libsuitesparseconfig.dylib
libgit2.dylib				libumfpack.5.7.8.dylib
libgmp.10.dylib				libumfpack.5.dylib
libgmp.dylib				libumfpack.dylib
libllvmcalltest.dylib			libz.1.2.11.dylib
libmbedcrypto.2.16.0.dylib		libz.1.dylib
libmbedcrypto.3.dylib			libz.dylib
libmbedcrypto.dylib			sys.dylib
libmbedtls.12.dylib			sys.dylib.dSYM
sec-12-0380:julia Mikael$ 
```
It is also not in my system library folder:
```
sec-12-0380:julia Mikael$ cd /usr/lib
sec-12-0380:lib Mikael$ ls
PN548_API.dylib				libiodbcinst.2.dylib
PN548_HAL_OSX.dylib			libiodbcinst.dylib
PN548_OSX.dylib				libipconfig.dylib
bundle1.o				libipsec.A.dylib
charset.alias				libipsec.dylib
closure					libk5crypto.dylib
cron					libkmod.a
crt1.10.5.o				libkmodc++.a
crt1.10.6.o				libkrb4.dylib
crt1.o					libkrb5.dylib
dtrace					libkrb524.dylib
dyld					libkrb5support.dylib
dylib1.10.5.o				libktrace.dylib
dylib1.o				libl.a
gcrt1.o					liblangid.dylib
groff					liblapack.dylib
lazydylib1.o				liblber.dylib
libATCommandStudio.a			libldap.dylib
libATCommandStudioDynamic.dylib		libldap_r.dylib
libAccountPolicyTranslation.dylib	liblzma.5.dylib
libAppletTranslationLibrary.dylib	liblzma.dylib
libAudioIssueDetector.dylib		libm.dylib
libAudioStatistics.dylib		libmarisa.dylib
libBSDPClient.A.dylib			libmecab.1.0.0.dylib
libBSDPClient.dylib			libmecab.dylib
libCRFSuite.dylib			libmecabra.dylib
libCRFSuite0.12.dylib			libmenu.5.4.dylib
libChineseTokenizer.dylib		libmenu.dylib
libCoreStorage.dylib			libmx.A.dylib
libDHCPServer.A.dylib			libmx.dylib
libDHCPServer.dylib			libncurses.5.4.dylib
libDiagnosticMessagesClient.dylib	libncurses.5.dylib
libFDR_osx.dylib			libncurses.dylib
libFosl_dynamic.dylib			libnetsnmp.15.1.2.dylib
libIASAuthReboot.dylib			libnetsnmp.15.dylib
libIASUnifiedProgress.dylib		libnetsnmp.25.dylib
libMatch.1.dylib			libnetsnmp.5.2.1.dylib
libMatch.dylib				libnetsnmp.5.dylib
libMaxEnt.dylib				libnetsnmp.dylib
libOpenScriptingUtil.dylib		libnetsnmpagent.25.dylib
libParallelCompression.dylib		libnetsnmpagent.dylib
libQMIParser.a				libnetsnmphelpers.25.dylib
libQMIParserDynamic.dylib		libnetsnmphelpers.dylib
libSMC.dylib				libnetsnmpmibs.25.dylib
libScreenReader.dylib			libnetsnmpmibs.dylib
libSystem.B.dylib			libnetsnmptrapd.25.dylib
libSystem.dylib				libnetsnmptrapd.dylib
libTelephonyUtilDynamic.dylib		libnetwork.dylib
libThaiTokenizer.dylib			libnetworkextension.dylib
libUniversalAccess.dylib		libnfrestore.dylib
libXSEvent.dylib			libnfshared.dylib
libXplugin.1.dylib			libobjc.A.dylib
libXplugin.dylib			libobjc.dylib
libapple_crypto.dylib			libodfde.dylib
libapple_nghttp2.dylib			libodmodule.dylib
libapr-1.0.dylib			libpam.1.dylib
libapr-1.dylib				libpam.2.dylib
libaprutil-1.0.dylib			libpam.dylib
libaprutil-1.dylib			libpanel.5.4.dylib
libarchive.2.dylib			libpanel.dylib
libarchive.dylib			libpcap.A.dylib
libate.dylib				libpcap.dylib
libauditd.0.dylib			libpcre.0.dylib
libauditd.dylib				libpcre.dylib
libauto.dylib				libpcreposix.0.dylib
libblas.dylib				libpcreposix.dylib
libboringssl.dylib			libpgtypes.3.4.dylib
libbsm.0.dylib				libpgtypes.3.dylib
libbsm.dylib				libpgtypes.dylib
libbz2.1.0.5.dylib			libpkstart.a
libbz2.1.0.dylib			libpmenergy.dylib
libbz2.dylib				libpmsample.dylib
libc++.1.dylib				libpoll.dylib
libc++.dylib				libpq.5.6.dylib
libc++abi.dylib				libpq.5.dylib
libc.dylib				libpq.dylib
libcblas.dylib				libprequelite.dylib
libcharset.1.0.0.dylib			libproc.dylib
libcharset.1.dylib			libpthread.dylib
libcharset.dylib			libpython.dylib
libclapack.dylib			libpython2.7.dylib
libcmph.dylib				libquit.dylib
libcom_err.dylib			libreadline.dylib
libcompression.dylib			libresolv.9.dylib
libcoreroutine.dylib			libresolv.dylib
libcoretls.dylib			librpcsvc.dylib
libcoretls_cfhelpers.dylib		libruby.2.3.0.dylib
libcrypto.0.9.7.dylib			libruby.2.3.dylib
libcrypto.0.9.8.dylib			libruby.dylib
libcrypto.35.dylib			libsandbox.1.dylib
libcrypto.41.dylib			libsandbox.dylib
libcrypto.dylib				libsasl2.2.0.1.dylib
libcsfde.dylib				libsasl2.2.0.15.dylib
libcups.2.dylib				libsasl2.2.0.21.dylib
libcups.dylib				libsasl2.2.0.22.dylib
libcupscgi.1.dylib			libsasl2.2.dylib
libcupscgi.dylib			libsasl2.dylib
libcupsimage.2.dylib			libspindump.dylib
libcupsimage.dylib			libsqlite3.0.dylib
libcupsmime.1.dylib			libsqlite3.dylib
libcupsmime.dylib			libssl.0.9.7.dylib
libcupsppdc.1.dylib			libssl.0.9.8.dylib
libcupsppdc.dylib			libssl.35.dylib
libcurl.3.dylib				libssl.43.dylib
libcurl.4.dylib				libssl.dylib
libcurl.dylib				libstdc++.6.0.9.dylib
libcurses.dylib				libstdc++.6.dylib
libdbm.dylib				libstdc++.dylib
libdes425.dylib				libsysdiagnose.dylib
libdl.dylib				libsysmon.dylib
libdns_services.dylib			libsystemstats.dylib
libdscsym.dylib				libtailspin.dylib
libdtrace.dylib				libtcl.dylib
libdz.dylib				libtcl8.5.dylib
libeasyperf.dylib			libtclstub8.5.a
libecpg.6.5.dylib			libtermcap.dylib
libecpg.6.dylib				libtidy.A.dylib
libecpg.dylib				libtidy.dylib
libecpg_compat.3.5.dylib		libtk.dylib
libecpg_compat.3.dylib			libtk8.5.dylib
libecpg_compat.dylib			libtkstub8.5.a
libedit.2.dylib				libtls.15.dylib
libedit.3.0.dylib			libtls.6.dylib
libedit.3.dylib				libtls.dylib
libedit.dylib				libtzupdate.dylib
libenergytrace.dylib			libusrtcp.dylib
libexpat.1.dylib			libutil.dylib
libexpat.dylib				libutil1.0.dylib
libexslt.0.dylib			libxar.1.dylib
libexslt.dylib				libxar.dylib
libextension.dylib			libxcselect.dylib
libf77lapack.dylib			libxml2.2.dylib
libffi.dylib				libxml2.dylib
libform.5.4.dylib			libxslt.1.dylib
libform.dylib				libxslt.dylib
libgcc_s.1.dylib			liby.a
libgcc_s.10.4.dylib			libz.1.1.3.dylib
libgcc_s.10.5.dylib			libz.1.2.11.dylib
libgermantok.dylib			libz.1.2.5.dylib
libgmalloc.dylib			libz.1.2.8.dylib
libgssapi_krb5.dylib			libz.1.dylib
libgutenprint.2.dylib			libz.dylib
libheimdal-asn1.dylib			log
libhunspell-1.2.0.0.0.dylib		pam
libhunspell-1.2.0.dylib			php
libhunspell-1.2.dylib			pkgconfig
libiconv.2.4.0.dylib			python2.7
libiconv.2.dylib			rpcsvc
libiconv.dylib				ruby
libicucore.A.dylib			sasl2
libicucore.dylib			sqlite3
libinfo.dylib				ssh-keychain.dylib
libiodbc.2.1.18.dylib			system
libiodbc.2.dylib			xpc
libiodbc.dylib				zsh
libiodbcinst.2.1.18.dylib
sec-12-0380:lib Mikael$ 
```

Two places to find it are 1. in a homebrewed GCC:
```
sec-12-0380:lib Mikael$ cd /usr/local/Cellar/gcc@8/8.3.0/lib/gcc/8
sec-12-0380:8 Mikael$ ls
gcc			libgfortran.a		libquadmath.a
libasan.5.dylib		libgfortran.dylib	libquadmath.dylib
libasan.dylib		libgfortran.spec	libsanitizer.spec
libasan_preinit.o	libgomp.1.dylib		libssp.0.dylib
libatomic.1.dylib	libgomp.a		libssp.a
libatomic.a		libgomp.dylib		libssp.dylib
libatomic.dylib		libgomp.spec		libssp_nonshared.a
libcc1.0.so		libitm.1.dylib		libstdc++.6.dylib
libcc1.so		libitm.a		libstdc++.a
libgcc_ext.10.4.dylib	libitm.dylib		libstdc++.a-gdb.py
libgcc_ext.10.5.dylib	libitm.spec		libstdc++.dylib
libgcc_s.1.dylib	libobjc-gnu.4.dylib	libstdc++fs.a
libgcc_s_ppc64.1.dylib	libobjc-gnu.a		libsupc++.a
libgcc_s_x86_64.1.dylib	libobjc-gnu.dylib	libubsan.1.dylib
libgfortran.5.dylib	libquadmath.0.dylib	libubsan.dylib
sec-12-0380:8 Mikael$ 
```
or 2. in the `CompilerSupportLibraries_jll` artifact:
```
sec-12-0380:8 Mikael$ cd /Users/Mikael/.julia/artifacts/0f0c9ee55a39f0c6ccc50e07baf798d64e0209cb/lib
sec-12-0380:lib Mikael$ ls
libasan.5.dylib		libgfortran.dylib	libquadmath.dylib
libasan.dylib		libgomp.1.dylib		libssp.0.dylib
libatomic.1.dylib	libgomp.dylib		libssp.dylib
libatomic.dylib		libitm.1.dylib		libstdc++.6.dylib
libgcc_s.1.dylib	libitm.dylib		libstdc++.dylib
libgcc_s_ppc64.1.dylib	libobjc-gnu.4.dylib	libubsan.1.dylib
libgcc_s_x86_64.1.dylib	libobjc-gnu.dylib	libubsan.dylib
libgfortran.5.dylib	libquadmath.0.dylib
sec-12-0380:lib Mikael$ 
```

When delivering binaries for Julia packages, one cannot assume that the end user has a homebrewed version of GCC, since they could have easily clicked on the `.dmg` from the download website rather than building from source. Therefore, if I am not mistaken, a macOS user would need to be able to load `CompilerSupportLibraries_jll` to obtain and locate `libgomp.1.dylib` in the most generic way possible.

Thanks for fixing `MPFR_jll`, btw!
